src/libmongoc/src/mongoc/mongoc-collection.c

```diff
@@ -3278,13 +3278,10 @@ mongoc_collection_find_and_modify_with_opts (
       }
 
       write_concern = appended_opts.writeConcern;
-   }
-
-   if (!write_concern) {
+   } else if (!write_concern) {
```

Comment:
Won't `write_concern` always be false here? The only line that could set this to true is `3281` but since this is an `else if` either `appended_opts.writeConcern` is `true` and line `3280` is reached or that condition is false in this block will be entered. That makes this essential an `else`.

Reply:
Yes! Good point.